sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -514,6 +515,13 @@ class DataFrameAggregateSuite extends QueryTest
     checkDataset(
       df.select(collect_set($"a"), collect_set($"b")).as[(Set[Int], Set[Int])],
       Seq(Set(1, 2, 3) -> Set(2, 4)): _*)
+
+    val bytesTest1 = "test1".getBytes
+    val bytesTest2 = "test2".getBytes
+    val df1 = Seq(bytesTest1, bytesTest1, bytesTest2).toDF("a")
+    val ret = df1.select(collect_set($"a")).collect()
+      .map(r => r.getAs[mutable.WrappedArray[_]](0)).head
```

Comment:
Please avoid casting things to their concrete implementation. This can break when we upgrade to newer scala version. In this case use `Seq[_]`.